### DISCUSSION TITLE
@Step and @Attachment works only with gradle, and with junit5 launcher don't work( Allure.step() works with junit5 launcher )

### DISCUSSION THREAD

**Discussion by mourav2222** [allure-framework/allure2#2063] [Questions & Support] [answered]
Why <code><b>@Step</b></code> don't wotk  with the junit5 launcher and only with gradle?
with gradle: gradlew clean test
![image](https://github.com/allure-framework/allure2/assets/76436807/dfce6cf3-1232-40a0-b359-b702c9d4c511)

with junit5 launcher: in Intellij Junit5 launcher
![image](https://github.com/allure-framework/allure2/assets/76436807/2ff571ce-b6ed-4a9d-9aad-43313a01d96a)

**Comment by baev** [comment]
You need to configure AspectJ for IDEA -- see https://www.jetbrains.com/help/idea/aspectj.html

**Reply by mourav2222** [reply]
@baev  
i start junit5 tests from comand line over junit launcher. The annotations don't work @Step and @Attachment from command line.
java -jar guitester.jar 
'Main-Class': 'de.cenit.phoenix.test.gui.cmd.Tester',

Can you say me, what i can add into build.gradle?

build.gradle
------------------------------
plugins {
    id 'io.qameta.allure' version '2.11.2'
    id 'java'
}

group = "de.cenit.phoenix"
version = "${version != 'unspecified' ? version : '1.0.0-SNAPSHOT'}".toString()

ext {
    allureToolVersion = '2.23.1'
    allureLibsVersion = '2.23.0'
    aspectjLibsVersion = '1.9.19'

}

java {
    sourceCompatibility = JavaVersion.VERSION_11
    targetCompatibility = JavaVersion.VERSION_11
}

repositories {
    mavenLocal()
    mavenCentral()
}

configurations {
    testImplementation {
        canBeResolved = true
    }
}

dependencies {

    // https://mvnrepository.com/artifact/org.slf4j/slf4j-api
    implementation "org.slf4j:slf4j-simple:${slf4jsimpleversion}"

    // test launcher
    testImplementation 'org.junit.platform:junit-platform-launcher:1.9.2'
    testImplementation platform('org.junit:junit-bom:5.9.1')


    annotationProcessor('info.picocli:picocli-codegen:4.7.1')
    testImplementation 'info.picocli:picocli:4.7.1'

    testImplementation 'commons-io:commons-io:2.11.0'
    testImplementation 'org.slf4j:slf4j-simple:${slf4jsimpleversion}'

    testImplementation 'org.junit.jupiter:junit-jupiter'

    // selenide-selenoid jar
    testImplementation "com.codeborne:selenide-selenoid:${selenideSelenoidversion}"

    // allure jars
    testImplementation "io.qameta.allure:allure-selenide:${allureLibsVersion}"
    testRuntimeOnly "org.aspectj:aspectjweaver:${aspectjLibsVersion}"


}

// Tasks: allureReport or allureServe
allure {
    version = allureToolVersion
    adapter {
        aspectjVersion = aspectjLibsVersion
    }
    report {
        version = allureLibsVersion
        reportDir = file('build/reports/allure-report')
    }
}


compileJava {
    options.warnings = true
    options.deprecation = false
    options.compilerArgs += ['-Xlint:unchecked', '-Xlint:deprecation']
}

shadowJar {
    archiveBaseName.set('guitester')
    archiveClassifier.set('')
    // Not set version in the jar name
    // archiveVersion.set('')

    manifest {
        attributes (
                'Main-Class': 'de.cenit.phoenix.test.gui.cmd.Tester',
                'Description': 'Application JAR file for gui tests execution on the command line'
        )
    }

    // Merge META-INF/services
    // /META-INF/services/org.openqa.selenium.remote.http.HttpClient$Factory
    // in selenium-remote-driver.jar and selenide-core.jar
    mergeServiceFiles()

    from sourceSets.test.output
    configurations = [project.configurations.testImplementation]

}

// gradlew createGuitesterJar
// copy /Y build\libs\guitester-5.5.11.0-000-SNAPSHOT.jar C:\Tools\guitester
// java -jar ./build/libs/guitester-5.5.11.0-000-SNAPSHOT.jar smoketest --tests=.*AdministrationUserInternalRoleAdmin.* --browser=firefox
tasks.register('createGuitesterJar') {
    dependsOn shadowJar
}


test {
    useJUnitPlatform() {

        if (project.hasProperty('test.skip')) {
            systemProperty 'guitests.skipguitest', true
        }

        // Remove src/test/resources/reportportal.properties file in jenkins job
        if (project.hasProperty('test.disablereportportal')) {
            systemProperty 'guitests.disablereportportal', true
        }
        if (project.hasProperty('test.video')) {
            systemProperty 'guitests.video', true
        }
        if (project.hasProperty('test.headless')) {
            systemProperty 'guitests.headless', true
        }
        if (project.hasProperty('test.browser')) {
            systemProperty 'guitests.browser', project.getProperty('test.browser')
        }
        if (project.hasProperty('test.version')) {
            systemProperty 'guitests.version', project.getProperty('test.version')
        }
        if (project.hasProperty('test.selenoid')) {
            systemProperty 'guitests.selenoid', project.getProperty('test.selenoid')
        }
        if (project.hasProperty('test.server')) {
            systemProperty 'guitests.server', project.getProperty('test.server')
        }
        if (project.hasProperty('test.browserbinary')) {
            systemProperty 'guitests.browserbinary', project.getProperty('test.browserbinary')
        }
        if (project.hasProperty('test.timeout')) {
            systemProperty 'guitests.timeout', project.getProperty('test.timeout')
        }
        if (project.hasProperty('test.screenshot')) {
            systemProperty 'guitests.screenshot', true
        }
        if (project.hasProperty('test.ignoreFailures')) {
            ignoreFailures = project.getProperty('test.ignoreFailures').toBoolean()
            println "ignoreFailures: " + ignoreFailures
        }
    }

    systemProperty 'junit.jupiter.extensions.autodetection.enabled', true

    // show standard out and standard error of the test JVM(s) on the console
    testLogging {
        showStandardStreams = true
        events = ["passed", "failed", "skipped"]
    }

    // set heap size for the test JVM(s)
    minHeapSize = "256m"
    maxHeapSize = "1g"

    // set JVM arguments for the test JVM(s)
    jvmArgs '-XX:MaxPermSize=256m'

    // listen to events in the test execution lifecycle
    beforeTest { desc ->
        println "Automation has started"
        logger.lifecycle("Running test: " + desc)
    }

    afterSuite { desc, result ->
        println "Automation has Finished - Results: ${result.resultType} (${result.testCount} tests, " +
                "${result.successfulTestCount} successes, ${result.failedTestCount} failures, ${result.skippedTestCount} skipped)"
    }

    // Fail the 'test' task on the first test failure
    // failFast = true

    // listen to standard out and standard error of the test JVM(s)
    onOutput { descriptor, event ->
        // logger.lifecycle("Test: " + descriptor + " produced standard out/err: " + event.message )
    }

}

test.onlyIf { !project.hasProperty('skipGuiTests') }

tasks.withType(Test) {
    /**
     * fix for retrofit https://github.com/square/retrofit/issues/3341
     */
    jvmArgs = ["--add-opens", "java.base/java.lang.invoke=ALL-UNNAMED"]
}

**Comment by mourav2222** [comment]
@baev i started from command line the  jar file with main function from the class Tester. I don't use gradle. And i don't get steps in the allure report wirh @Step only with Allure.step() function there are steps. Can you say me, why it happens and how can i fix this issue.
public class Tester {

    public static void main(String[] args) {

        int exitCode = new CommandLine(new MainCommand())
                 .setHelpFactory(new UnsortedSynopsisHelpFactory())
                 .setUsageHelpLongOptionsMaxWidth(30)
                 .execute(args); // |5|
        System.exit(exitCode);

    }

}

public class MainCommand implements Runnable {

    @Spec CommandSpec spec;
    @Override
    public void run() {
        throw new ParameterException(spec.commandLine(), "Specify a subcommand");
    }

}

**Comment by mourav2222** [comment, marked as answer]
@baev i found in the stackover of you https://stackoverflow.com/questions/51560256/allure-generate-allure-results-with-jar and
try the same with my jar file, where i use junit5 for tests and i only add to my command -javaagent and it works. Magic....
java -jar -javaagent:aspectjweaver-1.9.19.jar guitester-5.5.11.0-000-SNAPSHOT.jar %*

![image](https://github.com/allure-framework/allure2/assets/76436807/57c859a7-a52b-4831-a07e-d8a87e739570)
